libflux/src/semantic/nodes.rs

```diff
@@ -616,55 +616,79 @@ impl BinaryExpr {
-            | ast::Operator::PowerOperator
-            | ast::Operator::ModuloOperator => {
-                let mut constraints = vec![
-                    Constraint::Equal(self.left.type_of().clone(), self.typ.clone()),
-                    Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
-                ];
-                if let Some(kind) = match self.operator {
-                    ast::Operator::AdditionOperator => Some(Addable),
-                    ast::Operator::SubtractionOperator => Some(Subtractable),
-                    ast::Operator::MultiplicationOperator
-                    | ast::Operator::DivisionOperator
-                    | ast::Operator::PowerOperator
-                    | ast::Operator::ModuloOperator => Some(Divisible),
-                    _ => None,
-                } {
-                    constraints.push(Constraint::Kind(self.typ.clone(), kind));
-                }
-                Constraints::from(constraints)
-            }
-            // The following require the type to be a boolean.
-            ast::Operator::GreaterThanEqualOperator
-            | ast::Operator::LessThanEqualOperator
-            | ast::Operator::GreaterThanOperator
-            | ast::Operator::LessThanOperator
-            | ast::Operator::NotEqualOperator
-            | ast::Operator::EqualOperator => {
-                let kind = match self.operator {
-                    ast::Operator::EqualOperator | ast::Operator::NotEqualOperator => Equatable,
-                    _ => Comparable,
-                };
-                Constraints::from(vec![
-                    Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
-                    Constraint::Kind(self.left.type_of().clone(), kind),
-                    Constraint::Kind(self.right.type_of().clone(), kind),
-                    Constraint::Equal(self.typ.clone(), MonoType::Bool),
-                ])
-            }
+            ast::Operator::AdditionOperator => Constraints::from(vec![
+                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                Constraint::Equal(self.left.type_of().clone(), self.typ.clone()),
+                Constraint::Kind(self.typ.clone(), Kind::Addable),
+            ]),
+            ast::Operator::SubtractionOperator => Constraints::from(vec![
+                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                Constraint::Equal(self.left.type_of().clone(), self.typ.clone()),
+                Constraint::Kind(self.typ.clone(), Kind::Subtractable),
+            ]),
+            ast::Operator::MultiplicationOperator => Constraints::from(vec![
+                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                Constraint::Equal(self.left.type_of().clone(), self.typ.clone()),
+                Constraint::Kind(self.typ.clone(), Kind::Divisible),
```

Comment:
Could you explain this?

Reply:
I mean, why iuf you find a multiplication you require the Kind to be "Divisible"? Isn't it to much of a constraint? 🤔 

Ooook, but I see that `Int` for example is `Divisible`.

```
MonoType::Int => match with {
                Kind::Addable
                | Kind::Subtractable
                | Kind::Divisible
                | Kind::Comparable
                | Kind::Equatable
                | Kind::Nullable => Ok(Substitution::empty()),
            },
```

Reply:
Right now `Divisible` is the constraint for multiplication. Not sure if that will need to change in the future, but at least for right now every type that is divisible can also be multiplied. This is not true for `Addable` and `Subtractable` though, since strings are addable but not subtractable.